consensus/src/marshal/actor.rs

```diff
@@ -534,6 +531,33 @@ impl<
                                 }
                             }
                         }
+                        Message::SetFloor { height } => {
+                            if let Some(stored_height) = self.application_metadata.get(&LATEST_KEY) {
+                                if *stored_height >= height {
+                                    warn!(height, "sync floor not updated, lower than existing");
+                                    continue;
+                                }
+                            }
+
+                            if let Err(e) = self
+                                .application_metadata
+                                .put_sync(LATEST_KEY.clone(), height)
+                                .await
+                            {
+                                error!(?e, height, "failed to update sync floor");
+                                return;
+                            }
+                            self.last_processed_height = height;
+                            let _ = self.processed_height.try_set(self.last_processed_height);
+
+                            // Cancel any existing requests below the new sync floor.
+                            resolver.retain(Request::<B>::Finalized { height }.predicate()).await;
```

Comment:
`self::set_last_processed_height()` might be a good function that sets the value, updates the metadata, and retains the correct resolver requests